dotnet/ToDoApp/ToDoApp/ViewModel/ItemEntriesViewModel.cs

```diff
@@ -14,37 +15,99 @@ namespace ToDoApp
     public class ItemEntriesViewModel : INotifyPropertyChanged
     {
         private Realm _realm;
+        private IEnumerable<Item> _entries;
+        public IEnumerable<Item> Entries
+        {
+            get { return _entries; }
+            private set
+            {
+                if (_entries == value)
+                {
+                    return;
+                }
+                _entries = value;
+                PropertyChanged.Invoke(this, new PropertyChangedEventArgs(nameof(Entries)));
+            }
+        }
 
-        private INavigation _navigation;
-
-        public IEnumerable<Item> Entries { get; private set; }
-
-        public ICommand SignOutCommand { get; private set; }
+        public ICommand LogOutCommand { get; private set; }
 
         public ICommand DeleteEntryCommand { get; private set; }
 
         public ICommand AddEntryCommand { get; private set; }
 
         public event PropertyChangedEventHandler PropertyChanged;
 
-        public ItemEntriesViewModel(INavigation navigation, Realm realm)
+        public ItemEntriesViewModel()
         {
-            _navigation = navigation;
-            _realm = realm;
-
-            // Get the list of items.
-            Entries = _realm.All<Item>().OrderBy(i => i.Timestamp);
-
-            SignOutCommand = new Command(() =>
+            LogOutCommand = new Command(() =>
             {
+                if (User.Current == null)
+                {
+                    return;
+                }
                 User.Current.LogOutAsync().IgnoreResult();
-
-                // Go to welcome page.
-                _navigation.PopToRootAsync().IgnoreResult();
+                _realm = null;
+                Entries = null;
+                StartLoginCycle().IgnoreResult();
             });
 
             DeleteEntryCommand = new Command<Item>(DeleteEntry);
             AddEntryCommand = new Command(() => AddEntry().IgnoreResult());
+
+            StartLoginCycle().IgnoreResult();
+        }
+
+        private async Task StartLoginCycle()
+        {
+            do
+            {
+                await Task.Yield();
+            } while (await LogIn() == false);
+        }
+
+        private async Task<bool> LogIn()
+        {
+            try
+            {
+                var user = User.Current;
+                if (user == null)
+                {
+                    // Not already logged in.
+                    LoginResult loginResult;
+                    loginResult = await UserDialogs.Instance.LoginAsync("Log in", "Enter a username and password");
+
+                    if (!loginResult.Ok)
+                    {
+                        return false;
+                    }
+
+                    // Create credentials with the given username and password.
+                    // Specify whether we are registering a new user or not with `createUser`.
```

Comment:
That part of the comment seems outdated (i.e. we no longer distinguish between registering and simply logging in).